File: grading/3-benchmarks+codegen/valid/_selectionsort.py
def selection_sort(array, length):
    for i in xrange(0, length):
        min = array[i]
        min_index = i
        for j in xrange(i+1, length):
            if array[j] < min:
                min = array[j]
                min_index = j
        
        if min_index != i:
            array[min_index] = array[i]
            array[i] = min

def main():
    array = []
    length = 50000
    for i in xrange(0, length):
        array.append(length-i-1)

    selection_sort(array, length)

    for i in range(10):
        print array[i]

if __name__ == "__main__":
    main()

File: grading/3-benchmarks+codegen/valid/selectionsort.go
//~0
//~1
//~2
//~3
//~4
//~5
//~6
//~7
//~8
//~9
//&Python: 1 minutes 8.5 seconds
//&C++: 11.1 seconds

package main

// Sorts the array in increasing order
func selection_sort(array []int, length int) {
    for i := 0; i < length; i++ {
        min := array[i]
        min_index := i
        for j := i+1; j < length; j++ {
            if array[j] < min {
                min = array[j]
                min_index = j
            }
        }

        if min_index != i {
            // Swap i and min_index
            array[min_index] = array[i]
            array[i] = min
        }
    }
}

func main() {
    // Create the array to sort
    var array []int
    length := 50000
    for i := 0; i < length; i++ {
        array = append(array, length-i-1);
    }

    // Sort the array
    selection_sort(array, length)

    // Print the sorted array
    for i := 0; i < 10; i++ {
        println(array[i])
    }
}
